File: app/src/androidTest/java/pl/srw/billcalculator/di/TestApplicationModule.java
package pl.srw.billcalculator.di;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import pl.srw.billcalculator.data.ApplicationRepo;
import pl.srw.billcalculator.data.settings.PricesRepo;
import pl.srw.billcalculator.settings.prices.PgePrices;
import pl.srw.billcalculator.util.BillSelection;

@Module
class TestApplicationModule {

    private ApplicationComponent productionComponent;

    TestApplicationModule(ApplicationComponent productionComponent) {
        this.productionComponent = productionComponent;
    }

    @Provides
    @Singleton
    PgePrices providePgePrices() {
        return productionComponent.getPgePrices();
    }

    @Provides
    @Singleton
    ApplicationRepo provideApplicationRepo() {
        return productionComponent.getApplicationRepo();
    }

    @Provides
    @Singleton
    BillSelection provideBillSelection() {
        return productionComponent.getBillSelection();
    }

    @Provides
    @Singleton
    PricesRepo providePricesRepo() { return productionComponent.getPricesRepo(); }
}

File: app/src/androidTest/java/pl/srw/billcalculator/di/TestApplicationModule.kt
package pl.srw.billcalculator.di

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
internal class TestApplicationModule(private val productionComponent: ApplicationComponent) {

    @Provides @Singleton
    fun providePgePrices() = productionComponent.pgePrices

    @Provides @Singleton
    fun provideApplicationRepo() = productionComponent.applicationRepo

    @Provides @Singleton
    fun provideBillSelection() = productionComponent.billSelection

    @Provides @Singleton
    fun providePricesRepo() = productionComponent.pricesRepo

    @Provides @Singleton
    fun provideHistoryRepo() = productionComponent.historyRepo

    @Provides @Singleton
    fun applicationContext() = productionComponent.context
}
